test/videojs.ads.test.js

```diff
@@ -1,1025 +1,732 @@
+});
+
+QUnit.test('adserror in ads-ready transitions to content-playback', function(assert) {
+  assert.expect(5);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.trigger('adserror');
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A content-playback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'adserror', 'The reason for content-playback should have been adserror');
+});
+
+QUnit.test('adskip in ads-ready transitions to content-playback', function(assert) {
+  assert.expect(5);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.trigger('adskip');
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A content-playback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'adskip', 'The reason for content-playback should have been adskip');
+});
+
+QUnit.test('adserror in preroll? transitions to content-playback', function(assert) {
+  assert.expect(6);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.tech.trigger('play');
+  assert.strictEqual(this.player.ads.state, 'preroll?');
+  this.player.trigger('adserror');
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A content-playback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'adserror', 'The reason for content-playback should have been adserror');
+});
+
+QUnit.test('adskip in preroll? transitions to content-playback', function(assert) {
+  assert.expect(6);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.tech.trigger('play');
+  assert.strictEqual(this.player.ads.state, 'preroll?');
+  this.player.trigger('adskip');
+  assert.strictEqual(this.player.ads.state, 'content-playback');
+  assert.strictEqual(this.contentPlaybackSpy.callCount, 1, 'A content-playback event should have triggered');
+  assert.strictEqual(this.contentPlaybackReason(), 'adskip', 'The reason for content-playback should have been adskip');
+});
+
+QUnit.test('adserror in postroll? transitions to content-playback and fires ended', function(assert) {
+  assert.expect(8);
+  assert.strictEqual(this.player.ads.state, 'content-set');
+  this.player.trigger('adsready');
+  assert.strictEqual(this.player.ads.state, 'ads-ready');
+  this.player.tech.trigger('play');
+  this.player.trigger('adtimeout');
```

Comment:
A `setImmediate` callback was previously required here. Any idea why it's no longer necessary?